File: WebShopAppFrontend/web-shop-app/src/app/model/proizvodZahtjev.ts
export class ProizvodZahtjev {

  kratakNaslov: string;
  detaljanOpis: string;
  cijena: number;
  polovan: boolean;
  lokacija: string;
  kategorijaId: number;
  korisnickoIme: string

  constructor(
    kratakNaslov: string,
    detaljanOpis: string,
    cijena: number,
    polovan: boolean,
    lokacija: string,
    kategorijaId: number,
    korisnickoIme: string
  ) {
    this.kratakNaslov = kratakNaslov;
    this.detaljanOpis = detaljanOpis;
    this.cijena = cijena;
    this.polovan = polovan;
    this.lokacija = lokacija;
    this.kategorijaId = kategorijaId
    this.korisnickoIme = korisnickoIme;
  }

}

File: WebShopAppBackend/webshop/src/main/java/org/vs/dto/ProizvodZahtjev.java
package org.vs.dto;

import lombok.*;

import jakarta.validation.constraints.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ProizvodZahtjev {

	@NotBlank
	@Size(min = 3, max = 30)
	private String kratakNaslov;

	@NotBlank
	@Size(min = 3, max = 300)
	private String detaljanOpis;

	@Positive
	@Digits(integer = 10, fraction = 2)
	@NotNull
	private Double cijena;

	private boolean polovan;

	@Size(min = 2, max = 50)
	private String lokacija;

	@NotNull
	private Integer kategorijaId;

	@NotBlank
	@Pattern(regexp = "^[a-zA-Z0-9_]*$")
	@Size(min = 3, max = 100)
	private String korisnickoIme;

}
